providers/src/airflow/providers/celery/executors/celery_executor.py

```diff
@@ -450,7 +450,6 @@ def cleanup_stuck_queued_tasks(self, tis: list[TaskInstance]) -> list[str]:
         for ti in tis:
             readable_tis.append(repr(ti))
             task_instance_key = ti.key
-            self.fail(task_instance_key, None)
```

Comment:
This is problematic - folks could install the provider with this code onto 2.8.

Reply:
@jedcunningham can you explain a bit further? Is there a workaround we can consider? The reason we did this is because the idea of failing the task and then setting it to scheduled after felt fraught with potential bugs

Reply:
it's possible in the provider to inspect the airflow version and handle conditionally

Reply:
@dstandish yep added exactly that.